File: backend/internal/entities/language.go
package entities

import "time"

// Language represents a language in the system
type Language struct {
	ID        int       `json:"id"`
	Name      string    `json:"name"`
	CreatedAt time.Time `json:"created_at"`
}

// LanguageProficiency represents a language proficiency level
type LanguageProficiency struct {
	ID        int       `json:"id"`
	Name      string    `json:"name"`
	CreatedAt time.Time `json:"created_at"`
}

// UserLanguage represents a user's language proficiency
type UserLanguage struct {
	UserID        int                  `json:"user_id"`
	LanguageID    int                  `json:"language_id"`
	ProficiencyID int                  `json:"proficiency_id"`
	Language      *Language            `json:"language,omitempty"`
	Proficiency   *LanguageProficiency `json:"proficiency,omitempty"`
	CreatedAt     time.Time            `json:"created_at"`
}

File: frontend/src/types/language.ts
import { BaseEntity } from './common';

/**
 * Language-related types
 */

// Language model
export interface Language {
  id: number;
  name: string;
  created_at: string;
}

// Language proficiency level
export interface LanguageProficiency {
  id: number;
  name: string;
  created_at: string;
}

// User's language proficiency 
export interface UserLanguage {
  user_id: number;
  language_id: number;
  proficiency_id: number;
  language?: Language;
  proficiency?: LanguageProficiency;
  created_at: string;
}

// Language update for user profile
export interface UserLanguageUpdate {
  language_id: number;
  proficiency_id: number;
} 